File: 0020-valid-parentheses/0020-valid-parentheses.cpp
//Iterate over the characters
//if character is opening bracket then push into stack
//store top of the stack and compare if it is closing bracket or not
//return if stack is empty(valid) or not
class Solution {
public:
    bool isValid(string s) {
     stack<char>  stack;
     for(int i=0; i<s.length(); i++){
        if(s[i]=='(' || s[i]=='{' || s[i]=='['){
            stack.push(s[i]);
        }else{
            if(stack.empty()) return false;
            else{
                char top=stack.top();
                if((s[i] == ')' && top == '(') || (s[i] == '}' && top == '{') || (s[i] == ']' && top == '[')){
                    stack.pop();
                }else{
                    return false;
                }
            }
        }


     }
     return stack.empty();   
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
     Stack<Character> stack=new Stack<>();
     for(int i=0; i<s.length(); i++){
        if(s.charAt(i)=='(' || s.charAt(i)=='{' || s.charAt(i)=='['){
            stack.push(s.charAt(i));
        }else{
            char top=stack.peek();
            if((s.charAt(i)=='}' && top=='{') || (s.charAt(i)==']' && top=='[') || (s.charAt(i)==')' && top=='(')){
                stack.pop();
            }
        }
    }
    return stack.isEmpty();
}
}